File: app/distritos.py
from flask import Blueprint, jsonify
import requests

distritos_routes = Blueprint('distritos', __name__)

DGEG_BASE_URL = "https://precoscombustiveis.dgeg.gov.pt/api/PrecoComb"

@distritos_routes.route('/api/distritos', methods=['GET'])
def obter_distritos():
    try:
        response = requests.get(f"{DGEG_BASE_URL}/GetDistritos")
        data = response.json().get('resultado', [])
        return jsonify(data)
    except Exception as e:
        return jsonify({'erro': str(e)}), 500

File: frontend/src/api/distritos.ts
import axios from 'axios';

const API_URL = '/api/distritos';

export const getDistritos = async () => {
  try {
    const response = await axios.get(API_URL);
    return response.data;
  } catch (error) {
    console.error('Erro ao buscar distritos:', error);
    throw error;
  }
};


